### DISCUSSION TITLE
Pulp's removal of 'list_solvers' in v2.8.0 release broke snakemake

### DISCUSSION THREAD

**Discussion by corneliusroemer** [coin-or/pulp#721] [General]
I just wanted to let you know that the removal of `list_solvers` in a minor release broke e.g. snakemake, see https://github.com/snakemake/snakemake/issues/2607

Maybe this should have been a major release? Did you provide deprecation warnings since deprecation? I don't remember seeing them in snakemake but maybe snakemake silenced them.

**Comment by pchtsp** [comment]
Hello, you're right.

I honestly did not expect anyone to be using `list_solvers` or `get_solver` since this was documented only for a very brief time (or not documented at all).

As per deprecation warnings, you're right again: we usually added deprecation warnings for most things we removed but not for these specific functions. We will be more careful in the future.

**Reply by corneliusroemer** [reply]
Github code search is great for this! It does show that snakemake used it, see all the forks in code results

https://github.com/search?q=pulp.list_solvers+language%3APython&type=code

There are actually quite a few non-snakemake results as well

https://github.com/search?q=pulp.get_solver+language%3APython&type=code

And there might be more cases where it's not chained after `pulp.`
